src/GraphQL/Types/Scalars/Enumeration/EnumerationGraphType.cs

```diff
@@ -97,7 +97,9 @@ public void Add(EnumValueDefinition value)
         var foundByValue = Values.FindByValue(value);
 
         return foundByValue == null
-            ? value == null ? null : ThrowSerializationError(value)
+            ? value == null
+                ? null
+                : throw new InvalidOperationException($"Unable to serialize '{value}' value of type '{value.GetType().GetFriendlyName()}' to the enumeration type '{Name}'. Enumeration does not contain such value. Available values: {string.Join(", ", Values.Select(v => $"'{v.Value}' of type '{v.Value?.GetType().GetFriendlyName()}'"))}.")
```

Comment:
OK?

Reply:
It's okay.  Probably could be better, such as using `"DOG"` rather than `'DOG' of type 'String'` and `Bug1773Enum.Hello` rather than `'Hello' value of type 'Bug1773Enum'` -- seems a bit wordy now.  Feel free to merge or tweak further as desired.

Reply:
Ultimately this exception should typically only occur during development, and rarely during production.  So whether the exception message is perfect or not makes little difference so long as the programmer understands the issue and can fix it.  Meaning, 'whatever you think is fine' :smile:

Reply:
"DOG" is good but value may be of any type - that is why I added `of type....`

Reply:
The same for `'Hello' value of type 'Bug1773Enum'` instead of `Bug1773Enum.Hello` . I'll keep message as is for now.

Reply:
Quotes could denote a string like it does in C#

Reply:
> The same for `'Hello' value of type 'Bug1773Enum'` instead of `Bug1773Enum.Hello` . I'll keep message as is for now.

Ok

Reply:
> Quotes could denote a string like it does in C#

Of course, but special code is required to check type and concatenate proper text.